app/models/draft_content_item.rb

```diff
@@ -35,6 +35,29 @@ class DraftContentItem < ActiveRecord::Base
   }
   validates_with RoutesAndRedirectsValidator
 
+  validates :description, well_formed_content_types: { must_include: "text/html" }
+  validates :details, well_formed_content_types: { must_include: "text/html" }
+
+  # Postgres's JSON columns have problems storing literal strings, so these
+  # getter/setter overrides wrap and unwrap those strings in hashes.
+  def description=(value)
+    if value.is_a?(String)
+      super(string: value)
+    else
+      super
+    end
+  end
+
+  def description
+    value = super
+
+    if value.is_a?(Hash) && value.key?(:string)
+      value.fetch(:string)
+    else
+      value
+    end
+  end
```

Comment:
If anyone has ideas on how to make this better – I'd appreciate it.

I had a quick poke around [in Rails](https://github.com/rails/rails/blob/7f18ea14c893cb5c9f04d4fda9661126758332b5/activerecord/lib/active_record/type/internal/abstract_json.rb#L13), but stopped as it began to spiral into a rabbit hole.

I had a quick go at upgrading Rails to 4.2.5 (we're currently 4.2.4), but this didn't help. We should probably do this upgrade anyway. I may put in a pull request.